app/assets/javascripts/discourse/app/initializers/ember-events.js

```diff
@@ -1,43 +1,49 @@
 let initializedOnce = false;
 
+/**
+ * By default Ember listens to too many events. This tells it the only events
+ * we're interested in. (it removes mousemove, touchstart and touchmove)
+ *
+ * @type {Record<string, string>}
+ */
+export const EVENTS = {
+  touchend: "touchEnd",
+  touchcancel: "touchCancel",
+  keydown: "keyDown",
+  keyup: "keyUp",
+  keypress: "keyPress",
+  mousedown: "mouseDown",
+  mouseup: "mouseUp",
+  contextmenu: "contextMenu",
+  click: "click",
+  dblclick: "doubleClick",
+  focusin: "focusIn",
+  focusout: "focusOut",
+  mouseenter: "mouseEnter",
+  mouseleave: "mouseLeave",
+  submit: "submit",
+  input: "input",
+  change: "change",
+  dragstart: "dragStart",
+  drag: "drag",
+  dragenter: "dragEnter",
+  dragleave: "dragLeave",
+  dragover: "dragOver",
+  drop: "drop",
+  dragend: "dragEnd",
+};
+
 export default {
   name: "ember-events",
 
   initialize() {
-    // By default Ember listens to too many events. This tells it the only events
-    // we're interested in. (it removes mousemove, touchstart and touchmove)
     if (initializedOnce) {
       return;
     }
 
     // eslint-disable-next-line no-undef
     Ember.EventDispatcher.reopen({
-      events: {
-        touchend: "touchEnd",
-        touchcancel: "touchCancel",
-        keydown: "keyDown",
-        keyup: "keyUp",
-        keypress: "keyPress",
-        mousedown: "mouseDown",
-        mouseup: "mouseUp",
-        contextmenu: "contextMenu",
-        click: "click",
-        dblclick: "doubleClick",
-        focusin: "focusIn",
-        focusout: "focusOut",
-        mouseenter: "mouseEnter",
-        mouseleave: "mouseLeave",
-        submit: "submit",
-        input: "input",
-        change: "change",
-        dragstart: "dragStart",
-        drag: "drag",
-        dragenter: "dragEnter",
-        dragleave: "dragLeave",
-        dragover: "dragOver",
-        drop: "drop",
-        dragend: "dragEnd",
-      },
+      events: EVENTS,
```

Comment:
Do we still need this? In theory, no component-handleable events should be bubbling all the way up to the root app element, so we could remove all the app-level listeners? 

```suggestion
      events: {},
```
I tried this quickly and it made for some weird behaviour... which might be an indication that we didn't catch every type of event? 🤔

Reply:
Good callout. Let me dig further into this to better understand why and how the `EventDispatcher` is still in use.

Reply:
I've discovered that the set of events passed to the `EventDispatcher` can be trimmed down to a single event: `click`. I haven't yet discovered why this particular event is still necessary to delegate, but at least it should be much easier to investigate now that it's been isolated.